lib/types.nix

```diff
@@ -291,6 +291,15 @@ rec {
       functor = (defaultFunctor name) // { wrapped = elemType; };
     };
 
+    # Same as attrsOf, but lazy towards attribute values (at the cost of not supporting mkIf and etc)
+    lazyAttrsOf = elemType: attrsOf elemType // {
```

Comment:
I'm not sure if it's worth introducing this confusing and slightly module breaking type (because it makes `mkIf`, etc. not work anymore), just to add support for using the slightly deprecated `packageOverrides` as overlays. If people want overlays they can use overlays directly, if the package set they're using doesn't support this then that should be fixed there imo.